deerlab/snlls.py

```diff
@@ -284,6 +284,31 @@ def snlls(y, Amodel, par0, lb=None, ub=None, lbl=None, ubl=None, nnlsSolver='cvx
     regparam_prev = 0
     par_prev = [0]*len(par0)
 
+
+    def linear_problem(A,optimize_alpha):
+    #===========================================================================
+        """
+        Linear problem
+        ------------------
+        Solves the linear subproblem of the SNLLS objective function via linear LSQ 
+        constrained, unconstrained, or regularized.
+        """
+        global alpha
+        
+        # Regularization components
+        if optimize_alpha:

```

Comment:
The calls to `selregparam` and `lsqcomponents` (a few lines down) should only happen if regularization is requested.

Reply:
The call to `selregparam` happens only when regularization is requested (in which case `optimize_alpha=False`).  The logic statements are defined in `ResidualsFcn` in lines 328-351. 

The call to `lsqcomponents ` is required even when regularization is not requested. That function handles other things such as global optimization and proper preparation of the arguments of NNLS.  In the case that regularization is not needed, `alpha=0` and `L=eye(N.N)` so that the penalty is not included even though they are passed to `lsqcomponents `.

Reply:
The `alpha=0`and `L=eye(N,N)` then better goes into the defaults of the argument list of `lsqcomponents`.

Reply:
I did so. `lsqcomponents` now sets the `alpha` to zero and `L` to a diagonal matrix if not specified.